visa_mock/base/base_mocker.py

```diff
@@ -223,3 +223,7 @@ def send(self, scpi_string: str) -> Any:
 
 
 scpi = BaseMocker.scpi
+
+
+def reformat(scpi_string_pattern: str) -> str:
```

Comment:
Some docstring would be very welcome for this function. Explain this with examples.

Reply:
Also, a name like `compile_regular_expression` would be a better name. Say that, given a scpi string, a regular expression is generated. Again, we want examples as well.

Reply:
Detailed docstring has been added.

We are not giving a scpi string, but a scpi string **pattern**, and we are re-formatting the pattern so that re.match would do the correct job, so I prefer a name like `reformat_re_pattern` to `compile_re`. Thoughts?

Reply:
Add the following docstring: 

This function creates a regular expression pattern given a 
    SCPI string. This regular expression follows the SCPI 
    language rule where upper case letters denote the short form 
    of a SCPI command. A SCPI message send by software to the 
    instrument must match either the long or the short form.
    
    Examples: 
        >>> scpi = "VOLTage:CHANnel(.*) (.*)"  # From an instrument manual  
        >>> # the upper case part denotes the command short form
        >>> regex = compile_regular_expression(scpi)
        >>> # Sanity check to see if we match the long form
        >>> assert re.match(regex, "voltage:channel1 2.3").groups() == ('1', '2.3')
        >>> # Check if we match the shortform 
        >>> assert re.match(regex, "volt:chan1 2.3").groups() == ('1', '2.3')
        >>> # We should be able to mix and match 
        >>> assert re.match(regex, "voltage:chan1 2.3").groups() == ('1', '2.3')
    
    Args: 
        scpi: A string representing a pattern with which SCPI commands send by 
            software should match. These patterns can often be found in instrument 
            manuals.
    
    Returns: 
        A regex pattern which will match long and short forms. 
    
    Notes: 
        This function works by finding all groups of lower case letters which are 
        preceded by upper case letters. These groups of lower case letters are 
        marked as optional in the regex. Furthermore, this group is marked as 
        non-capturing. Please see 
        
        https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups
        
        Section "Non-capturing and Named Groups"